baseband/gsb/payload.py

```diff
@@ -126,9 +132,32 @@ def fromfile(cls, fh, payloadsize=None, bps=4, nchan=1,
                 for payload, part in zip(payload_set, thread):
                     part[:] = payload.words.reshape(-1, bpfs // 8)
 
-        return cls(words.ravel(), bps=bps, sample_shape=(nthread, nchan),
+        sample_shape_nthread = cls._sample_shape_cls_nthread(nthread, nchan)
+        return cls(words.ravel(), bps=bps, sample_shape=sample_shape_nthread,
                    complex_data=complex_data)
 
+    @classmethod
```

Comment:
And with the above, it should no longer be needed to overwrite `fromdata`

Reply:
Changed as per above.